components/datalog/Effectful/Zoo/DataLog/Data/Serialiser.hs

```diff
@@ -0,0 +1,32 @@
+module Effectful.Zoo.DataLog.Data.Serialiser
+  ( Serialiser(..),
+    handleLazyByteStringSerialiser,
+    stdoutLazyByteStringSerialiser,
+    stderrLazyByteStringSerialiser,
+  ) where
+
+import Data.ByteString.Lazy qualified as LBS
+import HaskellWorks.Prelude hiding (Floating(..))
+import System.IO (Handle)
+import System.IO qualified as IO
+
+data Serialiser i = Serialiser
+  { serialise :: i -> LBS.ByteString
+  , write :: LBS.ByteString -> IO ()
```

Comment:
I currently have the serialiser store an `write` `IO` function used to write logs.  Will I be able to provide this function in a `run` function.

I now want to define this function:

```
runDataLog :: ()
  => HasCallStack
  => (LBS.ByteString -> Eff r ())
  -> Eff (DataLog i : r) a
  -> Eff r a
runDataLog f = ...
```

The `f` needs to be converted into `LBS.ByteString -> IO ()` and put into the serialiser and the serialised passed into `withDataLogSerialiser`.

Is that possible?